### PR TITLE
rename lcs to bcs

### PR DESCRIPTION
**Pull request by bmwill**
Rename LCS to BCS.

I'm unsure if this rename will be a breaking change wrt to the changes in Move

### CODE REVIEW

**Comment by libra-action** [comment]
**This PR may have modified JSON-RPC server code.**

Breaking changes policy:

1. Chaning JSON-RPC method signatures or removing/modifying fields
   in the response in /V1 will not be accepted.
2. Adding fields is ok.
3. Adding a new JSON-RPC method is ok.

Please ensure you have also done the following to update the docs:

1. Update json-rpc/API-CHANGELOG.md.
2. Add/update type/method documentation under json-rpc/docs.

**Comment by libra-action** [comment]
**LEC dependency change summary**:
```
target packages:
  A bcs 0.1.1 (direct third-party, crates.io)
    * features: [none]
  R libra-canonical-serialization 0.1.0 (direct third-party, crates.io)
    * (old features: [none])
```

**Comment by libra-action** [comment]
**LSR dependency change summary**:
```
target packages:
  A bcs 0.1.1 (direct third-party, crates.io)
    * features: [none]
  R libra-canonical-serialization 0.1.0 (direct third-party, crates.io)
    * (old features: [none])
```

**Comment by libra-action** [comment]
**release binaries dependency change summary**:
```
target packages:
  A bcs 0.1.1 (direct third-party, crates.io)
    * features: [none]
  M libra-canonical-serialization 0.1.0 (transitive third-party, crates.io)
    * status changed from direct third-party
    * (unchanged features: [none])
```

**Comment by libra-action** [comment]
**LSR dependency change summary**:
```
target packages:
  A bcs 0.1.1 (direct third-party, crates.io)
    * features: [none]
  R libra-canonical-serialization 0.1.0 (direct third-party, crates.io)
    * (old features: [none])
```

**Comment by libra-action** [comment]
**LEC dependency change summary**:
```
target packages:
  A bcs 0.1.1 (direct third-party, crates.io)
    * features: [none]
  R libra-canonical-serialization 0.1.0 (direct third-party, crates.io)
    * (old features: [none])
```

**Comment by libra-action** [comment]
**release binaries dependency change summary**:
```
target packages:
  A bcs 0.1.1 (direct third-party, crates.io)
    * features: [none]
  M libra-canonical-serialization 0.1.0 (transitive third-party, crates.io)
    * status changed from direct third-party
    * (unchanged features: [none])
```

**Comment by libra-action** [comment]
**LEC dependency change summary**:
```
target packages:
  A bcs 0.1.1 (direct third-party, crates.io)
    * features: [none]
  R libra-canonical-serialization 0.1.0 (direct third-party, crates.io)
    * (old features: [none])
```

**Comment by libra-action** [comment]
**LSR dependency change summary**:
```
target packages:
  A bcs 0.1.1 (direct third-party, crates.io)
    * features: [none]
  R libra-canonical-serialization 0.1.0 (direct third-party, crates.io)
    * (old features: [none])
```

**Comment by libra-action** [comment]
**release binaries dependency change summary**:
```
target packages:
  A bcs 0.1.1 (direct third-party, crates.io)
    * features: [none]
  M libra-canonical-serialization 0.1.0 (transitive third-party, crates.io)
    * status changed from direct third-party
    * (unchanged features: [none])
```

**Comment by libra-action** [comment]
**LSR dependency change summary**:
```
target packages:
  A bcs 0.1.1 (direct third-party, crates.io)
    * features: [none]
  R libra-canonical-serialization 0.1.0 (direct third-party, crates.io)
    * (old features: [none])
```

**Comment by libra-action** [comment]
**LEC dependency change summary**:
```
target packages:
  A bcs 0.1.1 (direct third-party, crates.io)
    * features: [none]
  R libra-canonical-serialization 0.1.0 (direct third-party, crates.io)
    * (old features: [none])
```

**Comment by libra-action** [comment]
**release binaries dependency change summary**:
```
target packages:
  A bcs 0.1.1 (direct third-party, crates.io)
    * features: [none]
  M libra-canonical-serialization 0.1.0 (transitive third-party, crates.io)
    * status changed from direct third-party
    * (unchanged features: [none])
```

File: specifications/README.md [hunk 1]
```diff
@@ -53,7 +53,7 @@ As [depicted](#The-Diem-network), there are 4 network protocol specifications:
 
 In addition, these specifications build on top of a common set of specifications:
 
-* **[Diem canonical serialization (LCS)](https://developers.diem.com/docs/rustdocs/diem_canonical_serialization/)**. This details how data is canonically serialized and deserialized. Hashing and signing of data structures are applied after LCS serialization.
+* **[Diem canonical serialization (BCS)](https://developers.diem.com/docs/rustdocs/diem_canonical_serialization/)**. This details how data is canonically serialized and deserialized. Hashing and signing of data structures are applied after BCS serialization.
```
Comment (bob-wilson) [inline specifications/README.md:56]: Should the doc file in the URL be named to .../rustdocs/binary_canonical_serialization?

File: specifications/consensus/README.md [hunk 2]
```diff
@@ -12,7 +12,7 @@ This document is organized as follows:
 4. [Abstracted modules](#Abstracted-modules) - The components this specification depends on.
 5. [Consensus modules](#Consensus-modules) - The components built upon common data structures that are described as a part of this specification.
 
-All network communication occurs over DiemNet and any serialization, deserialization and hashing is determined by [LCS](https://developers.diem.com/docs/rustdocs/diem_canonical_serialization/).
+All network communication occurs over DiemNet and any serialization, deserialization and hashing is determined by [BCS](https://developers.diem.com/docs/rustdocs/diem_canonical_serialization/).
```
Comment (bob-wilson) [inline specifications/consensus/README.md:15]: Ditto here.

File: specifications/crypto/README.md [hunk 3]
```diff
@@ -157,7 +157,7 @@ We will look at how the domain separator defined in this section is used
 ## Serialization
 
 Serialization of a payload is done using
-[LCS](https://developers.diem.com/docs/rustdocs/diem_canonical_serialization/). This
+[BCS](https://developers.diem.com/docs/rustdocs/diem_canonical_serialization/). This
```
Comment (bob-wilson) [inline specifications/crypto/README.md:160]: ...and here

**Review by bob-wilson — COMMENTED** [review]
This looks pretty good. I found a few small things that should get fixed but it's not urgent to do that right away.

**Comment by bmwill** [comment]
@bob-wilson Thanks for the review. I'll make sure those changes are made

**Comment by libra-action** [comment]
**LEC dependency change summary**:
```
target packages:
  A bcs 0.1.1 (direct third-party, crates.io)
    * features: [none]
  R libra-canonical-serialization 0.1.0 (direct third-party, crates.io)
    * (old features: [none])
```

**Comment by libra-action** [comment]
**release binaries dependency change summary**:
```
target packages:
  A bcs 0.1.1 (direct third-party, crates.io)
    * features: [none]
  M serde-generate 0.17.0 (direct third-party, crates.io)
    * version upgraded from 0.16.1
    * (unchanged features: [none])
  R libra-canonical-serialization 0.1.0 (direct third-party, crates.io)
    * (old features: [none])
```

**Comment by libra-action** [comment]
**LSR dependency change summary**:
```
target packages:
  A bcs 0.1.1 (direct third-party, crates.io)
    * features: [none]
  R libra-canonical-serialization 0.1.0 (direct third-party, crates.io)
    * (old features: [none])
```

**Comment by libra-action** [comment]
**LSR dependency change summary**:
```
target packages:
  A bcs 0.1.1 (direct third-party, crates.io)
    * features: [none]
  R libra-canonical-serialization 0.1.0 (direct third-party, crates.io)
    * (old features: [none])
```

**Comment by libra-action** [comment]
**release binaries dependency change summary**:
```
target packages:
  A bcs 0.1.1 (direct third-party, crates.io)
    * features: [none]
  M serde-generate 0.17.0 (direct third-party, crates.io)
    * version upgraded from 0.16.1
    * (unchanged features: [none])
  R libra-canonical-serialization 0.1.0 (direct third-party, crates.io)
    * (old features: [none])
```

**Comment by libra-action** [comment]
**LEC dependency change summary**:
```
target packages:
  A bcs 0.1.1 (direct third-party, crates.io)
    * features: [none]
  R libra-canonical-serialization 0.1.0 (direct third-party, crates.io)
    * (old features: [none])
```

**Comment by libra-action** [comment]
**LEC dependency change summary**:
```
target packages:
  A bcs 0.1.1 (direct third-party, crates.io)
    * features: [none]
  R libra-canonical-serialization 0.1.0 (direct third-party, crates.io)
    * (old features: [none])
```

**Comment by libra-action** [comment]
**LSR dependency change summary**:
```
target packages:
  A bcs 0.1.1 (direct third-party, crates.io)
    * features: [none]
  R libra-canonical-serialization 0.1.0 (direct third-party, crates.io)
    * (old features: [none])
```

**Comment by libra-action** [comment]
**release binaries dependency change summary**:
```
target packages:
  A bcs 0.1.1 (direct third-party, crates.io)
    * features: [none]
  M serde-generate 0.17.0 (direct third-party, crates.io)
    * version upgraded from 0.16.1
    * (unchanged features: [none])
  R libra-canonical-serialization 0.1.0 (direct third-party, crates.io)
    * (old features: [none])
```

**Comment by libra-action** [comment]
**LSR dependency change summary**:
```
target packages:
  A bcs 0.1.1 (direct third-party, crates.io)
    * features: [none]
  R libra-canonical-serialization 0.1.0 (direct third-party, crates.io)
    * (old features: [none])
```

**Comment by libra-action** [comment]
**LEC dependency change summary**:
```
target packages:
  A bcs 0.1.1 (direct third-party, crates.io)
    * features: [none]
  R libra-canonical-serialization 0.1.0 (direct third-party, crates.io)
    * (old features: [none])
```

**Comment by libra-action** [comment]
**release binaries dependency change summary**:
```
target packages:
  A bcs 0.1.1 (direct third-party, crates.io)
    * features: [none]
  M serde-generate 0.17.0 (direct third-party, crates.io)
    * version upgraded from 0.16.1
    * (unchanged features: [none])
  R libra-canonical-serialization 0.1.0 (direct third-party, crates.io)
    * (old features: [none])
```

**Comment by libra-action** [comment]
**LEC dependency change summary**:
```
target packages:
  A bcs 0.1.1 (direct third-party, crates.io)
    * features: [none]
  R libra-canonical-serialization 0.1.0 (direct third-party, crates.io)
    * (old features: [none])
```

**Comment by libra-action** [comment]
**release binaries dependency change summary**:
```
target packages:
  A bcs 0.1.1 (direct third-party, crates.io)
    * features: [none]
  M serde-generate 0.17.0 (direct third-party, crates.io)
    * version upgraded from 0.16.1
    * (unchanged features: [none])
  R libra-canonical-serialization 0.1.0 (direct third-party, crates.io)
    * (old features: [none])
```

**Comment by libra-action** [comment]
**LSR dependency change summary**:
```
target packages:
  A bcs 0.1.1 (direct third-party, crates.io)
    * features: [none]
  R libra-canonical-serialization 0.1.0 (direct third-party, crates.io)
    * (old features: [none])
```

**Review by sherry-x — APPROVED** [review]

**Pull request merged**